server/service/src/invoice_line/stock_in_line/generate.rs

```diff
@@ -0,0 +1,130 @@
+use repository::{
+    InvoiceLineRow, RepositoryError, StockLineRow, StockLineRowRepository, StorageConnection,
+};
+use util::uuid::uuid;
+
+pub fn convert_stock_line_to_single_pack(stock_line: StockLineRow) -> StockLineRow {
+    StockLineRow {
+        total_number_of_packs: stock_line.total_number_of_packs * stock_line.pack_size as f64,
+        available_number_of_packs: stock_line.available_number_of_packs
+            * stock_line.pack_size as f64,
+        cost_price_per_pack: stock_line.cost_price_per_pack / stock_line.pack_size as f64,
+        sell_price_per_pack: stock_line.sell_price_per_pack / stock_line.pack_size as f64,
+        pack_size: 1,
+        ..stock_line
+    }
+}
+
+pub fn convert_invoice_line_to_single_pack(invoice_line: InvoiceLineRow) -> InvoiceLineRow {
+    InvoiceLineRow {
+        number_of_packs: invoice_line.number_of_packs * invoice_line.pack_size as f64,
+        sell_price_per_pack: invoice_line.sell_price_per_pack / invoice_line.pack_size as f64,
+        cost_price_per_pack: invoice_line.cost_price_per_pack / invoice_line.pack_size as f64,
+        pack_size: 1,
+        ..invoice_line
+    }
+}
+pub struct StockLineInput {
+    pub stock_line_id: Option<String>,
+    pub store_id: String,
+    pub on_hold: bool,
+    pub barcode_id: Option<String>,
+    pub supplier_link_id: String,
+    pub overwrite_stock_levels: bool,
+}
+
+struct StockLevels {
+    available_number_of_packs: f64,
+    total_number_of_packs: f64,
+}
+
+pub fn generate_batch(
```

Comment:
This generate method has been somewhat refactored - sorry has been moved from mod file so a bit hard to compare 😭